XForm/XForm/Functions/Coalesce.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using XForm.Data;
+using XForm.Query;
+
+namespace XForm.Functions
+{
+    public class Coalesce : IFunctionBuilder
+    {
+        public Type ReturnType => null;
+
+        public string Usage => "Coalesce({Col|Func|Const}, ...)";
+
+        public string Name => "Coalesce";
+
+        public IXColumn Build(IXTable source, XDatabaseContext context)
+        {
+            return new CoalesceTransformFunction(source, context);
+        }
+    }
+
+    public class CoalesceTransformFunction : IXColumn
+    {
+        private List<IXColumn> _columns = new List<IXColumn>();
+        private ICoalescer _coalescer;
+
+        public CoalesceTransformFunction(IXTable source, XDatabaseContext context)
+        {
+            while (context.Parser.HasAnotherArgument)
+            {
+                IXColumn column = context.Parser.NextColumn(source, context);
+                _columns.Add(column);
+                if (System.String.IsNullOrEmpty(column.ColumnDetails.Name)) throw new ArgumentException($"Column {_columns.Count} passed to 'Column' wasn't assigned a name. Use 'AS [Name]' to assign names to every column selected.");
```

Comment:
Why are you checking that every input has a name? It seems like you don't need that particular restriction, and many functions may not assign names to their results.